sdk/azidentity/chained_token_credential.go

```diff
@@ -14,15 +14,20 @@ import (
 
 // ChainedTokenCredentialOptions contains optional parameters for ChainedTokenCredential.
 type ChainedTokenCredentialOptions struct {
-	// placeholder for future options
+	// If true, it will not assume the first successful credential should be always used.
+	RetryAllSources bool
 }
 
 // ChainedTokenCredential is a chain of credentials that enables fallback behavior when a credential can't authenticate.
 type ChainedTokenCredential struct {
-	sources []azcore.TokenCredential
+	sources              []azcore.TokenCredential
+	successfulCredential azcore.TokenCredential
+	retryAllSources      bool
 }
 
 // NewChainedTokenCredential creates a ChainedTokenCredential.
+// By default, this credential will assume that the first successful credential should be the only credential used on future requests.
+// If the `RetryAllSources` option is set to true, it will always try to get a token using all of the available credentials.
```

Comment:
```suggestion
// If the `RetryAllSources` option is set to true, it will always try to get a token using all of the originally provided credentials.
```